### PR TITLE
A couple CI improvements

### PR DESCRIPTION
**Pull request by apoelstra**
Bump MSRV and add fuzztests to CI.

### CODE REVIEW

**Comment by sanket1729** [comment]
Okay, I can see how clippy is really annoying here. The same code that passed Clippy checks in 2020 is failing now. I guess they added more checks. 

Seems like fuzz is already broken in the existing code

**Comment by apoelstra** [comment]
Fixed clippy and fuzztest compilation. But the fuzztests fail now (`rtt_program` fails with `6b0f7f09000000000000000eff5500`) so I guess my compilation test fixes were incorrect :)

**Comment by sanket1729** [comment]
I will investigate into this issue

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -75,3 +75,35 @@ jobs:
...
           command: clippy
           args: -- -D warnings
+
+  fuzz:
+    name: Fuzz Tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - nightly
```
Comment (sanket1729) [inline .github/workflows/main.yml:85]: Hongfuzz frequently breaks on nightly, so all repos in the rust-bitcoin ecosystem switched to stable for fuzzing. The CI also fails to download dependencies in nghtly here. I think we should also change this to stable

**Comment by sanket1729** [comment]
I may have never ran the fuzz tests again. And this is probably some of my old code causing the crash. Looking into it

**Comment by apoelstra** [comment]
@sanket1729 I think this is ready to review now. It seems that the fuzz test failures were quite simple bugs in the code.

**Comment by apoelstra** [comment]
Lol, sigh, now honggfuzz isn't compiling on stable.

**Comment by sanket1729** [comment]
Interesting, I think rust-bitcoin, rust-miniscript both use hongfuzz on stable now.

**Comment by sanket1729** [comment]
I am able to build it locally on 1.58 ..

**Comment by sanket1729** [comment]
@apoelstra, you need to install these. 
See rust-miniscript actions file for reference : https://github.com/rust-bitcoin/rust-miniscript/blob/b8d04accb1e5fcf02849113531783a860ef42bbc/.github/workflows/rust.yml#L16-L17

**Comment by apoelstra** [comment]
CI passes :)

**Review by sanket1729 — APPROVED** [review]
ACK 31724a30be054ffc445b12250bdccb377270352e. We can have another PR to update these with C implementation

**Pull request merged**